### PR TITLE
Release/4.6.0

### LINKED ISSUE #86: Release version 4.6.0
This issue is for tracking the 4.6.0 release. Target release date: **March 25 2024**

## Pre-release steps

- [x] Merge #62  
- [x] Merge #67  
- [x] Merge #71   
- [x] Merge #70  
- [x] Merge #75   
- [x] Merge #79  
- [x] Merge #82   

## Release steps

- [x] Branch: Starting from `trunk`, cut a release branch named `release/4.6.0`
- [x] Version bump: Bump the version number in `woocommerce-square.php`, `package.json`, `package-lock.json` and `readme.txt` to 4.6.0. In `woocommerce-square.php` update both the plugin "Version:" property and the plugin `WC_SQUARE_VERSION` constant
- [x] Update `@since`: Find all new `@since x.x.x` / `@since n.e.x.t` lines and update those with the new version number, 4.6.0
- [x] Generate changelog: Locally, generate the changelog by running the following Woorelease command: `php woorelease.phar cl:generate --product_version=4.6.0 https://github.com/woocommerce/woocommerce-square/tree/trunk`. Ensure the changelog updates look correct
- [x] Update changelog: Take the changelog generated from the previous step and update the release PR with those changes, both in the `changelog.txt` file and `readme.txt` file
- [x] Open PR: Open a new PR against `trunk` from the release branch, `release/4.6.0`
- [x] Run tests: On this release PR, add the label `needs: qit default tests` and `needs: e2e testing`. This will trigger the default QIT tests (activation, API and E2E) and our E2E tests. Ensure these tests pass as well as all other applicable tests
- [x] Merge: Merge the release PR into `trunk`
- [x] Prepare ZIP: Prepare the zip by running Woorelease locally on the `trunk` branch: `php woorelease.phar build https://github.com/woocommerce/woocommerce-square/tree/trunk`. Make sure you are running the proper versions of `node`, `npm` and `composer` to ensure the built files are consistent across releases. Versions can be found in [`package.json`](https://github.com/woocommerce/woocommerce-square/blob/trunk/package.json)
- [x] Compare: Download the latest [released zip](https://github.com/woocommerce/woocommerce-square/releases) and then run a diff compare against the generated zip, ensuring all expected changes are there (and no unexpected changes)
- [x] Test ZIP: Take the ZIP generated from the previous step and test this locally, ensuring version numbers look correct, the changelog looks correct and the plugin installs and activates properly
- [x] Attach ZIP: Attach the ZIP to this issue and ping the Automattic team in the `#a8c-10up-woo-support` channel to complete the release

### PR DESCRIPTION
**Pull request by dkotter**
### Changes proposed in this Pull Request:

Version bump and prep for the 4.6.0 release

Closes #86 

### Steps to test the changes in this Pull Request:

Verify all file changes look accurate

### CODE REVIEW

**Comment by dkotter** [comment]
@vikrampm1 E2E tests are failing here (the environment isn't starting correctly, seems WooCommerce isn't being installed). That will need looked at before proceeding with this release

**Comment by iamdharmesh** [comment]
@dkotter E2E test is fixed now and ready for the next steps.

cc: @vikrampm1

**Review by vikrampm1 — APPROVED** [review]

File: changelog.txt [hunk 1]
```diff
@@ -1,6 +1,15 @@
 *** WooCommerce Square Changelog ***
 
-= 4.5.2 - TBD =
+= 4.6.0 - 2024-03-25 =
```
Comment (dkotter) [inline changelog.txt]:
```suggestion
= 4.6.0 - 2024-03-27 =
```

File: readme.txt [hunk 2]
```diff
@@ -75,7 +75,16 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
 
 == Changelog ==
 
-= 4.5.2 - TBD =
+= 4.6.0 - 2024-03-25 =
```
Comment (dkotter) [inline readme.txt]:
```suggestion
= 4.6.0 - 2024-03-27 =
```

**Pull request merged**